feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/views/priceCharts/ChartController.kt

```diff
@@ -0,0 +1,165 @@
+    }
+
+    private val context = chart.context
+
+    private var currentEntries: List<Entry> = emptyList()
+    private var useNeutralColor = false
+
+    init {
+        setupChartUI()
+    }
+
+    fun showYAxis(show: Boolean) {
+        chart.axisRight.setDrawLabels(show)
+        chart.invalidate()
+    }
+
+    fun useNeutralColor(useNeutral: Boolean) {
+        this.useNeutralColor = useNeutral
+        updateChart()
+    }
+
+    fun setEntries(entries: List<Entry>) {
+        currentEntries = entries
+        updateChart()
+    }
+
+    private fun setupChartUI() {
+        chart.setBackgroundColor(Color.TRANSPARENT)
+        chart.setDrawGridBackground(false)
+        chart.setDrawBorders(false)
+
+        chart.description.isEnabled = false
+        chart.legend.isEnabled = false
+        chart.axisLeft.isEnabled = false
+        chart.xAxis.isEnabled = false
+        chart.setScaleEnabled(false)
+        chart.minOffset = 0f
+        chart.extraTopOffset = 12f
+        chart.extraBottomOffset = 12f
+
+        chart.renderer = PriceChartRenderer(
+            highlightColor = context.getColor(R.color.neutral_price_chart_line),
+            dotRadius = 4.dpF(context),
+            strokeWidth = 4.dpF(context),
+            strokeAlpha = 0.16f,
+            chart = chart
+        )
+
+        chart.axisRight.apply {
+            isEnabled = true
+            textSize = 9f
+            textColor = context.getColor(R.color.text_secondary)
+            typeface = Typeface.MONOSPACE
+            setLabelCount(4, true)
+            setDrawTopYLabelEntry(true)
+            gridLineWidth = 1.5f
+            setDrawAxisLine(false)
+            setDrawGridLines(true)
+            gridColor = context.getColor(R.color.price_chart_grid_line)
+            enableGridDashedLine(10f, 10f, 0f)
+        }
+
+        chart.marker = null
+        chart.onChartGestureListener = PriceChartGestureListener(
+            onGestureStart = {
+                val highlight = chart.getHighlightByTouchPoint(it.x, it.y)
+                chart.highlightValue(highlight)
+                val entry = chart.data?.getEntryForHighlight(highlight)
+                entry?.let { updateChartWithSelectedEntry(it) }
+            },
+            onGestureEnd = {
+                chart.highlightValue(null)
+                updateChart()
+            }
+        )
+
+        chart.setOnChartValueSelectedListener(object : OnChartValueSelectedListener {
+            override fun onValueSelected(e: Entry, h: Highlight) {
+                updateChartWithSelectedEntry(e)
+            }
+
+            override fun onNothingSelected() {
+                updateChart()
+            }
+        })
+    }
+
+    private fun updateChartWithSelectedEntry(entry: Entry) {
+        val entriesBefore = currentEntries.filter { it.x <= entry.x }
+        val entriesAfter = currentEntries.subList(entriesBefore.size - 1, currentEntries.size)
```

Comment:
`val (entriesBefore, entriesAfter) = currentEntries.partition { it.x <= entry.x }`